packages/calling-component-bindings/src/participantListSelectorUtils.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CommunicationIdentifier, getIdentifierKind } from '@azure/communication-common';
+import { memoizeFnAll, toFlatCommunicationIdentifier } from '@internal/acs-ui-common';
+import { CallParticipantListParticipant } from '@internal/react-components';
+
+/**
+ * @private
+ */
+export const memoizedConvertAllremoteParticipants = memoizeFnAll(
+  (
```

Comment:
To make it safe, the first param of the memoizeFunction should be key(the key to find the right memoized instance), and the key is flatten parameter like string or number (Object is okay, but we will never know when it updates)

Let's flatten the identifier first then use it as the first parameter outside of the function